File: api/src/main/java/com/hirrao/javaee/service/UserService.java
package com.hirrao.javaee.service;


import com.hirrao.javaee.entity.User;
import com.baomidou.mybatisplus.core.metadata.IPage;
import org.springframework.stereotype.Service;

@Service
public interface UserService {
    User findByUsername(String userName);

    User findByUid(long uid);

    void register(long uid, String userName, String userPassword, String phoneNumber);

    void update(long uid, String phoneNumber, String sex, String birthday);

    void updatePassword(long uid, String userPassword);

    IPage<User> accountsInfo(int curPage,int size);

    void modifyUserInfo(long uid, String userName,String phoneNumber, String sex, String birthday,int permission);

    void updateUserName(long uid, String userName);
    void updateSex(long uid, String sex);
    void updateBirthday(long uid, String birthday);

    void deleteUser(long uid);

    IPage<User> searchUserByCondition(int curPage,int size,String searchCondition,String conditionValue);
}

File: api/src/main/kotlin/cn/hirrao/javaee/service/UserService.kt
package cn.hirrao.javaee.service

import cn.hirrao.javaee.entity.User
import com.baomidou.mybatisplus.core.metadata.IPage
import org.springframework.stereotype.Service


@Service
interface UserService {
    fun findByUsername(userName: String?): User?

    fun findByUid(uid: Int): User?

    fun findByPhoneNumber(phoneNumber: String?): User?

    fun register(uid: Int, userName: String?, userPassword: String?, phoneNumber: String?)

    fun update(uid: Int, phoneNumber: String?, sex: String?, birthday: String?)

    fun updatePassword(uid: Int, userPassword: String?)

    fun accountsInfo(curPage: Int, size: Int): IPage<User?>?

    fun modifyUserInfo(
        uid: Long,
        userName: String?,
        phoneNumber: String?,
        sex: String?,
        birthday: String?,
        permission: Int
    )

    fun updateUserName(uid: Int, userName: String?)
    fun updateSex(uid: Int, sex: String?)
    fun updateBirthday(uid: Int, birthday: String?)

    fun deleteUser(uid: Long)

    fun searchUserByCondition(curPage: Int, size: Int, searchCondition: String?, conditionValue: String?): IPage<User?>?
}
